WordPress/AbstractThemeSniff.php

```diff
@@ -0,0 +1,1352 @@
+		}
+		if ( false !== strpos( $file_content, 'get_the_post_thumbnail' ) ) {
+			$sniff_helper['post_thumbnail_support'] = true;
+		}
+		if ( false !== strpos( $file_content, 'get_post_thumbnail_id' ) ) {
+			$sniff_helper['post_thumbnail_support'] = true;
+		}
+	}
+
+	/**
+	 * Check that post tags are supported in the theme. At least one of the
+	 * following functions would need to be found in at least one of the template
+	 * files, fail if none are found at all; the_tags(), get_the_tag_list(),
+	 * get_the_term_list()
+	 * If one is present set flag to true for use later.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_post_tags_check( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, 'the_tags' ) ) {
+			$sniff_helper['post_tags_support'] = true;
+		}
+		if ( false !== strpos( $file_content, 'get_the_tag_list' ) ) {
+			$sniff_helper['post_tags_support'] = true;
+		}
+		if ( false !== strpos( $file_content, 'get_the_term_list' ) ) {
+			$sniff_helper['post_tags_support'] = true;
+		}
+	}
+
+	/**
+	 * Check that add_theme_support( 'title-tag' ) and wp_title() are being used.
+	 * If used then set a flag to true, to be used later.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_title_tag_check( $file_content ) {
+		global $sniff_helper;
+		if ( ! preg_match( '#add_theme_support\s?\(\s?[\'|"]title-tag#', $file_content ) ) {
+			$sniff_helper['title_tag']['theme_support'] = true;
+		}
+		if ( false !== strpos( $file_content, 'wp_title' ) ) {
+			$sniff_helper['title_tag']['wp_title'] = true;
+		}
+	}
+
+	/**
+	 * Check if register_sidebar(), dynamic_sidebar() and
+	 * add_action( 'widget_init', ... ) are being used.
+	 * If used then set a flag to true, to be used later.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_sidebar_checks( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, 'register_sidebar' ) ) {
+			$sniff_helper['sidebar_support']['register_sidebar_used'] = true;
+		}
+		if ( false !== strpos( $file_content, 'dynamic_sidebar' ) ) {
+			$sniff_helper['sidebar_support']['dynamic_sidebar_used'] = true;
+		}
+		// Don't use strict, for some reason you get false positives if strict is used.
+		if ( false != preg_match( '/add_action\s*\(\s*("|\')widgets_init("|\')\s*,/', $file_content ) ) {
+			$sniff_helper['sidebar_support']['widgets_init_used'] = true;
+		}
+	}
+
+	/**
+	 * Check that basic WordPress function calls are made.
+	 * If used then set a flag to true, to be used later.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_basic_function_checks( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, 'wp_footer' ) ) {
+			$sniff_helper['basic_function_calls']['wp_footer'] = true;
```

Comment:
Yes, I need to do a sniff to check that it is in the correct place. This check is a onve through to see if it is even there. It will be used in the regular sniff.